Wire U5, the deviatoric-only tet, into ccx.

Adds the dispatcher branches, the build entries and the matrix-structure
registration for elements_ccx/e_c3d_u5.f and resultsmech_u5.f.

mastruct.c and mastructcs.c each carry their own hard-coded list of user
element types, separate from e_c3d_u.f's dispatcher. An unregistered type
falls through to the substructure path and ccx dies looking for a
superelement matrix file -- the same trap U4 hit.

--- e_c3d_u.f	2026-08-22 11:20:26.682935314 +0300
+++ e_c3d_u.f	2026-08-22 13:44:25.560508157 +0300
@@ -88,6 +88,38 @@
      &       ne0,ipkon,thicke,integerglob,doubleglob,tieset,istartset,
      &       iendset,ialset,ntie,nasym,ielprop,prop)
 !        
+      elseif(lakonl(2:2).eq.'5') then
+!
+!       SPAX: U5, deviatoric-only tet (nodal-averaged B-bar)
+!
+        call e_c3d_u5(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
+     &       ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
+     &       alzero,ielmat,ielorien,norien,orab,ntmat_,
+     &       t0,t1,ithermal,vold,iperturb,nelemload,
+     &       sideload,xload,nload,idist,sti,stx,iexpl,plicon,
+     &       nplicon,plkcon,nplkcon,xstiff,npmat_,dtime,
+     &       matname,mi,ncmat_,mass,stiffness,buckling,rhsi,intscheme,
+     &       ttime,time,istep,iinc,coriolis,xloadold,reltime,
+     &       ipompc,nodempc,coefmpc,nmpc,ikmpc,ilmpc,veold,
+     &       ne0,ipkon,thicke,integerglob,doubleglob,tieset,istartset,
+     &       iendset,ialset,ntie,nasym,ielprop,prop)
+!
+      elseif(lakonl(2:2).eq.'4') then
+!
+!       SPAX: U4, mixed displacement/pressure tetrahedron (MINI, P1+bubble/P1)
+!
+        call e_c3d_u4(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
+     &       ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
+     &       alzero,ielmat,ielorien,norien,orab,ntmat_,
+     &       t0,t1,ithermal,vold,iperturb,nelemload,
+     &       sideload,xload,nload,idist,sti,stx,iexpl,plicon,
+     &       nplicon,plkcon,nplkcon,xstiff,npmat_,dtime,
+     &       matname,mi,ncmat_,mass,stiffness,buckling,rhsi,intscheme,
+     &       ttime,time,istep,iinc,coriolis,xloadold,reltime,
+     &       ipompc,nodempc,coefmpc,nmpc,ikmpc,ilmpc,veold,
+     &       ne0,ipkon,thicke,integerglob,doubleglob,tieset,istartset,
+     &       iendset,ialset,ntie,nasym,ielprop,prop)
+!
       elseif(lakonl(2:4).eq.'S45') then
         call e_c3d_us45(co,kon,lakonl,p1,p2,omx,bodyfx,nbody,s,sm,
      &       ff,nelem,nmethod,elcon,nelcon,rhcon,nrhcon,alcon,nalcon,
--- resultsmech_u.f	2026-08-22 11:20:26.684935397 +0300
+++ resultsmech_u.f	2026-08-22 13:44:25.568975239 +0300
@@ -68,6 +68,24 @@
      &        ncmat_,nstate_,stiini,vini,ener,eei,enerini,istep,iinc,
      &        reltime,calcul_fn,calcul_qa,calcul_cauchy,nener,
      &        ikin,nal,ne0,thicke,emeini,nelem,ielprop,prop,t0g,t1g)
+      elseif(lakon(nelem)(2:2).eq.'5') then
+!
+!     SPAX: U5 deviatoric tet
+!
+         call resultsmech_u5(co,kon,ipkon,lakon,v,stx,elcon,nelcon,
+     &        rhcon,nrhcon,alcon,nalcon,alzero,ielmat,ielorien,norien,
+     &        orab,ntmat_,t0,t1,ithermal,iperturb,fn,iout,qa,vold,
+     &        nmethod,dtime,plicon,nplicon,plkcon,nplkcon,xstiff,
+     &        npmat_,matname,mi,ncmat_,calcul_fn,calcul_qa,nal,nelem)
+      elseif(lakon(nelem)(2:2).eq.'4') then
+!
+!     SPAX: U4 mixed displacement/pressure tetrahedron
+!
+         call resultsmech_u4(co,kon,ipkon,lakon,v,stx,elcon,nelcon,
+     &        rhcon,nrhcon,alcon,nalcon,alzero,ielmat,ielorien,norien,
+     &        orab,ntmat_,t0,t1,ithermal,iperturb,fn,iout,qa,vold,
+     &        nmethod,dtime,plicon,nplicon,plkcon,nplkcon,xstiff,
+     &        npmat_,matname,mi,ncmat_,calcul_fn,calcul_qa,nal,nelem)
       elseif(lakon(nelem)(2:4).eq.'S45') then
 !
 !     user element us45 flat shell element with 4 nodes (5dof)
--- mastruct.c	2026-08-22 11:24:54.127557241 +0300
+++ mastruct.c	2026-08-22 13:46:32.511932403 +0300
@@ -127,6 +127,8 @@
 	indexe=ipkon[i];
 	if(strcmp1(&lakon[8*i],"U")==0){
 	  if((strcmp1(&lakon[8*i+1],"1")==0)||
+	     (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
 	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	     (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
@@ -358,7 +360,9 @@
       indexe=ipkon[i];
       if(strcmp1(&lakon[8*i],"U")==0){
 	if((strcmp1(&lakon[8*i+1],"1")==0)||
-	   (strcmp1(&lakon[8*i+1],"S45")==0)||
+	   (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	   (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
 	  /* user element
--- mastructcs.c	2026-08-22 11:24:54.128557286 +0300
+++ mastructcs.c	2026-08-22 13:46:32.512621269 +0300
@@ -69,7 +69,9 @@
     indexe=ipkon[i];
     if(strcmp1(&lakon[8*i],"U")==0){
       if((strcmp1(&lakon[8*i+1],"1")==0)||
-	 (strcmp1(&lakon[8*i+1],"S45")==0)||
+	 (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	 (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
 	/* user element
@@ -184,7 +186,9 @@
     indexe=ipkon[i];
     if(strcmp1(&lakon[8*i],"U")==0){
       if((strcmp1(&lakon[8*i+1],"1")==0)||
-	 (strcmp1(&lakon[8*i+1],"S45")==0)||
+	 (strcmp1(&lakon[8*i+1],"4")==0)||
+	     (strcmp1(&lakon[8*i+1],"5")==0)||
+	     (strcmp1(&lakon[8*i+1],"S45")==0)||
 	 (strcmp1(&lakon[8*i+1],"S3")==0)){
 	  
 	/* user element
--- Makefile.inc	2026-08-22 11:20:50.209913322 +0300
+++ Makefile.inc	2026-08-22 13:44:25.569853307 +0300
@@ -275,6 +275,11 @@
 e_c3d_se.f \
 e_c3d_th.f \
 e_c3d_u1.f \
+	e_c3d_u4.f \
+	e_c3d_u5.f \
+	resultsmech_u5.f \
+	u4mat.f \
+	resultsmech_u4.f \
 e_c3d_us3.f \
 e_c3d_us45.f \
 e_c3d_u.f \
